### PR TITLE
LPS-138756

### PR DESCRIPTION
**Pull request by arthurchan35**

### CODE REVIEW

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
Hey @arthurchan35 . I'm a bit unsure about this latest approach. The part about defaulting to RS256 when it is not configured. Because some existing deployments will break. Those with:
* Don't use discovery endpoint
* Have listed signing algorithms, and RS256 is not the first one listed. For example [ HS256, RS256 ]

Those currently work, but in future will fail SSO.

I'm a bit in two minds, because the help text for that field clearly states that RS256 should always be the first algorithm listed (so it should be fine). But realistically speaking, the only way to current support providers that send ID tokens using other algorithms would be to ignore that instruction! Thoughts?

**Comment by arthurchan35** [comment]
> Hey @arthurchan35 . I'm a bit unsure about this latest approach. The part about defaulting to RS256 when it is not configured. Because some existing deployments will break. Those with:
> 
> * Don't use discovery endpoint
> * Have listed signing algorithms, and RS256 is not the first one listed. For example [ HS256, RS256 ]
> 
> Those currently work, but in future will fail SSO.
> 
> I'm a bit in two minds, because the help text for that field clearly states that RS256 should always be the first algorithm listed (so it should be fine). But realistically speaking, the only way to current support providers that send ID tokens using other algorithms would be to ignore that instruction! Thoughts?

Hey Stian,

The 2 cases you listed have nothing to do with breaking current deployments.
The changes I made will potentially break in such case:

- Portal admin does not specify an algorithm in the new configuration entry, and registers a non-RS256 algorithm at provider site.


However as long as customers(portal admin) know about the change(when they receive it) and adds the registered algorithm to new configuration entry, they will be fine.

I have considered this problem when making the commit, the reason I decided to go this way:
1. It's the right implementation, per specs.
2. Customers were doing it wrong(I understand it was because of our wrong implementation), they need to correct it with our help(ie. fix pack changes notes etc).
3. In reality, the expected amount of customers who are using non-RS256 should be extremely little. In another word, the impact is minimal.
4. The only other alternative approach I could think of by then is that we use previous logic of using first algorithms from provider supported algorithm, as the default behaivor. I don't like this approach, because existing wrong doing won't be corrected if they choose not to.

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -7180,7 +7180,7 @@ register-a-fido2-authenticator=Register a FIDO2 Authenticator
 registered-app-urls=Registered Remote App URLs
 registered-date=Registered Date
 registered-id-token-signing-alg=Registered ID Token Signing Algorithm
-registered-id-token-signing-alg-help=Set the ID token signing algorithm that is registered at OIDC provider site. The value must be one of the supported algorithms specified in discovery endpoint, or configuration entry of ID Token Signing Algorithms. The default, if omitted, is RS256.
+registered-id-token-signing-alg-help=Set the ID token signing algorithm registered with the OpenId Connect provider. The default is RS256.
```
Comment (arthurchan35) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:7183]: The purpose of the description is not about validating, but about reminding portal admin what's the configuration is for.

I am okay if you want to reduce the despcriton.

**Comment by stian-sigvartsen** [comment]
Agreed breaking change. Document and sent via liferay-appsec/liferay-portal/pull/501

**Pull request closed without merging**